tests/databases/value/mongo_db/test_value_update.py

```diff
@@ -0,0 +1,233 @@
+import stk
+import pymongo
+
+from .utilities import (
+    get_database_state,
+)
+from ...utilities import (
+    DatabaseEntry,
+    DatabaseState,
+    assert_database_state,
+)
+
+
+def test_update_1():
+    """
+    Test that existing entries are updated.
+
+    """
+
+    collection = '_test_update_1'
+    database_name = '_test_update_1'
+    client = pymongo.MongoClient()
+    client.drop_database(database_name)
+
+    database = stk.ValueMongoDb(
+        mongo_client=client,
+        collection=collection,
+        database=database_name,
+        put_lru_cache_size=0,
+        get_lru_cache_size=0,
+    )
+
+    molecule = stk.BuildingBlock('CCC')
+
+    database.put(molecule, 12)
+    assert_database_state(
+        state1=get_database_state(database),
+        state2=DatabaseState({
+            DatabaseEntry(
+                InChIKey=stk.InchiKey().get_key(molecule),
+                v=12,
+            ): 1,
+        }),
+    )
+
+    database.put(molecule, 43)
+    assert_database_state(
+        state1=get_database_state(database),
+        state2=DatabaseState({
+            DatabaseEntry(
+                InChIKey=stk.InchiKey().get_key(molecule),
+                v=43,
+            ): 1,
+        }),
+    )
```

Comment:
Aha so the DatabaseState and DatabaseEntry methods are primarily used for testing?

Reply:
I assume that when assert_database_state is called, all of the other information stored in the database that are associated with that molecule are also asserted equal? Such as the atom and bond information

Reply:
Indeed, it asserts that the database has the exact molecules, atoms bonds, position matrices etc. It asserts the entire database state is exactly equal to what I expect.

Reply:
However - note that the ValueDatabase does NOT store molecules, atoms, bonds or position matrices. So in this case what you said is false. Because this information is not in the database  - nor is it expected to be.